nixos/modules/services/admin/nomachine.nix

```diff
@@ -0,0 +1,314 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.services.nxserver;
+
+  settingsFmt = rec {
+    type = with lib.types;
+      (attrsOf (nullOr (oneOf [str int bool attrs])))
+      // {
+        description = "settings option";
+      };
+
+    format = value:
+      lib.generators.toKeyValue {
+        mkKeyValue = k: v:
+          if lib.isAttrs v
+          then "Section \"${k}\"\n" + (format v) + "EndSection"
+          else if lib.isBool v
+          then
+            lib.generators.mkKeyValueDefault {} " " k (
+              if v == true
+              then 1
+              else 0
+            )
+          else if lib.isString v
+          then
+            lib.generators.mkKeyValueDefault {} " " k (
+              if k == "Name"
+              then "\"${v}\""
+              else v
+            )
+          else lib.generators.mkKeyValueDefault {} " " k v;
+      }
+      value;
+
+    generate = name: value:
+      pkgs.writeText name (
+        format value
+      );
+  };
+in {
+  options.services.nxserver = with lib; {
+    enable = mkEnableOption "the NoMachine remote desktop server";
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to open port 4000 in the firewall.
```

Comment:
The port can be different of 4000

Reply:
I think it's better to omit the port here

Reply:
Ah I forgot to change it, done now :)